openlibrary/plugins/upstream/code.py

```diff
@@ -271,10 +272,13 @@ def GET(self, key):
     def POST(self, key):
         i = web.input("v", _comment=None)
         v = i.v and safeint(i.v, None)
+
         if v is None:
             raise web.seeother(web.changequery({}))
 
-        if not web.ctx.site.can_write(key):
+        user = accounts.get_current_user()
+        admin = user and user.key in [m.key for m in web.ctx.site.get('/usergroup/admin').members]
```

Comment:
Here and below, how about using `is_admin` instead of `admin` so the variable role is reflected more clearly in its name?